.github/dependabot.yml

```diff
@@ -13,24 +13,52 @@ updates:
     # We don't trigger Jenkins or GitHub Actions builds on pull requests from dependabot,
     # so we can safely use a high limit here.
     open-pull-requests-limit: 20
+    groups:
+      # This group combines all build-only dependencies. Published artifacts do not depend on them.
+      # Grouping such dependencies will make Dependabot create PRs with a branch name
+      # following the pattern (`dependabot/maven/build-dependencies-.*`)
+      # and with a title like `Bump the build-dependencies group with 8 updates` that we can easily
+      # use for Hibernate Bot rules.
+      build-dependencies:
+        patterns:
+          # Maven plugin patterns:
+          - "*plugin*"
+          - "*maven*"
+          - "*surefire*"
+          - "*forbiddenapis*"
+          - "*owasp*"
+          # DB drivers:
+          - "com.h2database:h2"
+          - "org.apache.derby:derby"
+          - "org.postgresql:postgresql"
+          - "org.mariadb.jdbc:mariadb-java-client"
+          - "com.mysql:mysql-connector-j"
+          - "com.ibm.db2:jcc"
+          - "com.oracle.database.jdbc:ojdbc*"
+          - "com.microsoft.sqlserver:mssql-jdbc"
+          # Other test dependencies
+          - "org.apache.groovy:groovy-jsr223" # used for scripting maven plugin
+          - "org.apache.commons:commons-lang3" # used in hibernate-search-util-common tests
+          - "org.apache.commons:commons-math3" # used to solve dependency convergence for Wiremock
+          - "org.openjdk.jmh:*" # performance testing dependency
+          - "com.google.guava:guava" # Guava is used in our test utils
+          - "org.asciidoctor:*" # Asciidoctor is used for rendering the documentation
+          - "org.codehaus.plexus:*" # Eclipse compiler
+          - "org.jboss.marshalling:jboss-marshalling" # JBeret IT dependency
+          - "org.wildfly.security:wildfly-security-manager" # JBeret IT dependency
     ignore:
       # These dependencies are updated manually
       - dependency-name: "org.hibernate:*"
       - dependency-name: "org.hibernate.*:*"
       # AWS SDK releases way too often (every week?); ignore all patch updates
       - dependency-name: "software.amazon.awssdk:*"
         update-types: ["version-update:semver-patch"]
-      # JRuby releases way too often (every two weeks?) and is only used during the build; ignore all patch updates
-      - dependency-name: "org.jruby:jruby-complete"
-        update-types: ["version-update:semver-patch"]
       # Groovy is only used during the build; ignore all patch updates
       - dependency-name: "org.apache.groovy:groovy-jsr223"
         update-types: [ "version-update:semver-patch" ]
       # We don't care that much about being on the very latest version of some integration test dependencies
       - dependency-name: "org.springframework.boot:*"
         update-types: [ "version-update:semver-patch" ]
```

Comment:
Shouldn't this move to the `build-dependencies` group as well? Spring really is a build dependency, we don't actually publish any artifact that depends on it.

Reply:
hmm, aaaaaactually .... https://github.com/marko-bekhta/hibernate-search/pull/157
it seems like Dependabot is trying to upgrade the spring version where we don't want it to be upgraded ... maybe let's ignore spring upgrades in general?

Reply:
That's annoying. Would using a separate property help? I.e. define `                <version.org.springframework.boot.v2>2.7.14</version.org.springframework.boot.v2>` in the build-parent POM and use ` <version.org.springframework.boot>${version.org.springframework.boot.v2}</version.org.springframework.boot>` in the profile... maybe that'd confuse dependabot in just the right way?

Otherwise, sure, +1 to ignore spring upgrades if there's no workaround.

Reply:
or I just drop that profile 😄 
this was coming from the profile in common utils, but since we've ended up running tests in a spring IT module we weren't actually rebuilding common-utils sooooo, that should work too 😃 🤞🏻

Reply:
Good point!